### PR TITLE
changes in integrated spec KE flux in SpectralAnalysis.pyx and corresponding .pxd

### PR DESCRIPTION
**Pull request by jschroettle**
please check for consistency, current longer test is running, already ran for 4 days

### CODE REVIEW

File: SpectralAnalysis.pyx [hunk 1]
```diff
@@ -52,70 +71,94 @@ cdef class SpectralAnalysis:
...
                 uak = Gr.SphericalGrid.grdtospec(np.multiply(DV.Wp.values[:,:,k], DV.U.values[:,:,k]))
                 vak = Gr.SphericalGrid.grdtospec(np.multiply(DV.Wp.values[:,:,k], DV.V.values[:,:,k]))
                 uakD = Gr.SphericalGrid.grdtospec(np.multiply(DV.Wp.values[:,:,k], DV.U.values[:,:,k-1]))
                 vakD = Gr.SphericalGrid.grdtospec(np.multiply(DV.Wp.values[:,:,k], DV.V.values[:,:,k-1]))
                 u_specD = Gr.SphericalGrid.grdtospec(DV.U.values.base[:,:,k-1])
                 v_specD = Gr.SphericalGrid.grdtospec(DV.V.values.base[:,:,k-1])
+                #build discretized ke
+                u2_spec  = 0.5*np.multiply(uak, np.conj(u_spec))
+                v2_spec  = 0.5*np.multiply(vak, np.conj(v_spec))
+                u2_specD = 0.5*np.multiply(uakD, np.conj(u_specD))
```
Comment (yairchn) [inline SpectralAnalysis.pyx:117]: lets replace the `D` at the end of names like `u2_specD` with something more appropriate such as:
`u2_spec_dn`,
in general lets keep the CamelCase (using capital letters and no underscores) for classes and structs and use function_name that uses underscore for temporary variables. And more over I would not mix the two in the same name

File: SpectralAnalysis.pyx [hunk 2]
```diff
@@ -24,25 +24,44 @@ cdef class SpectralAnalysis:
...
         cdef:
             Py_ssize_t k, i
             Py_ssize_t nl = Pr.n_layers
-            double factor = self.spectral_frequency/(TS.t_max - self.spinup_time)
-            double complex [:] u_spec, v_spec, E_spec,  uak, vak
-            double [:,:] dp, dps_dx, dps_dy, dudx, dudy, dvdx, dvdy
+            double factor = self.spectral_frequency/(TS.t_max - self.spinup_time) # normalization with number of timesteps
+            double complex [:] uak, vak, uakD, vakD # temporary fields in spectral space
+            double complex [:] E_spec_adv, E_spec_div, E_spec_mass, E_spec_surf_corr, E_spec_flux_corr, E_spec_ps_corr # temporary fields in spectral space
+            #double [:] E_spec_adv_k, E_spec_div_k, E_spec_mass_k, E_spec_surf_corr_k, E_spec_flux_corr_k, E_spec_ps_corr_k # temporary array for functions of wavenumber
```
Comment (yairchn) [inline SpectralAnalysis.pyx:52]: should we delete this commented line?

File: SpectralAnalysis.pyx [hunk 3]
```diff
@@ -52,70 +71,94 @@ cdef class SpectralAnalysis:
             dudx, dudy  = Gr.SphericalGrid.getgrad(u_spec.base) # get gradients in grid space
             dvdx, dvdy  = Gr.SphericalGrid.getgrad(v_spec.base) # get gradients in grid space
 
-            # Kinetic Energy flux due to advection and divergence (first two terms in Appendix B1)
+            # Kinetic Energy flux due to advection (first term in Appendix B1)
             U_adv = np.add(np.multiply(DV.U.values[:,:,k],dudx), np.multiply(DV.V.values[:,:,k],dudy))
             V_adv = np.add(np.multiply(DV.U.values[:,:,k],dvdx), np.multiply(DV.V.values[:,:,k],dvdy))
+            uak = Gr.SphericalGrid.grdtospec(np.multiply(dp, U_adv))
```
Comment (yairchn) [inline SpectralAnalysis.pyx:77]: maybe we can come up with a better name here?
how about `mU_adv`

**Review by yairchn — COMMENTED** [review]
looks great - maybe we can consider some name changes see comments

File: SpectralAnalysis.pyx (hunk 1 above)
Comment (jschroettle) [reply SpectralAnalysis.pyx:117]: good idea

File: SpectralAnalysis.pyx (hunk 2 above)
Comment (jschroettle) [reply SpectralAnalysis.pyx:52]: yes

File: SpectralAnalysis.pyx (hunk 3 above)
Comment (jschroettle) [reply SpectralAnalysis.pyx:77]: that's super!

**Pull request merged**